### DISCUSSION TITLE
Error occurred prerendering page "/". ReferenceError: self is not defined

### DISCUSSION THREAD

**Discussion by kamlendras** [vercel/next.js#66799] [Help] [answered]
### Summary

Link to the code that reproduces this issue
[](https://github.com/kamlendras/OpenCryptoExchange)

To Reproduce
npm install / yarn install / bun install
npm run build / yarn run build / bun run build
![20240612_18h49m07s_grim](https://github.com/vercel/next.js/assets/96082996/454dd763-275a-4d04-881d-882ab2fa7519)
Its been so many days i am unable to solved this issue. Also i read https://nextjs.org/docs/messages/prerender-error but its not helping. :sad

### Additional information

```js
Current vs. Expected behavior
Project should succesfully deploy , not is not able to deploy.

Provide environment information
OS : Arch Linux
"dependencies": {
    "@emotion/react": "^11.11.4",
    "@emotion/styled": "^11.11.5",
    "@fontsource/inter": "^5.0.18",
    "@mui/icons-material": "^5.15.19",
    "@mui/joy": "^5.0.0-beta.36",
    "add": "^2.0.6",
    "next": "14.2.3",
    "react": "^18",
    "react-animated-numbers": "^0.18.0",
    "react-dom": "^18",
    "react-transition-group": "^4.4.5"
  },
  "devDependencies": {
    "typescript": "^5",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "14.2.3"
  }
```

### Example

https://github.com/kamlendras/OpenCryptoExchange

**Comment by icyJoseph** [comment, marked as answer]
I suspect, a component is trying to render, assuming it is running only in the browser, but in SSR world, that's not true. Maybe try removing parts of the `/` route, until it builds, then you'd be able to tell which library you are using is not behaving correctly.

**Comment by BeatrizGivisiez** [comment]
only rename docs
<img width="237" alt="fix" src="https://github.com/user-attachments/assets/698db83b-bb6f-4187-ba6e-2b76bdc6d6ff">

**Reply by Islamosama1** [reply]
It is work to me , Thanks
